APEXMALHAR-2314 Improper functioning in partitioning for sequentialFileRead for FSRecord

deepak-narkhede opened the pull request:
Fix the StreamCodec for FSRecordReader, initially it was hashcode of blockId's mostly always unique.
Hence unable to satisfy the sequentialFileRead property. Now the StreamCodec is modified to work
with hashcode of filePath. So all blocks related to a file would be partitioned on same operator.

Tested with recordReader and verified for sequentialFileRead that all blocks related to a file are partitioned to single operator.

yogidevendra reviewed (changes requested):
Could you please shorten the commit message?
Keep details to 1-2 lines per change.

Additional details can be captured in the JIRA description. Git log showing long messages would be hard to read.

deepak-narkhede committed:
APEXMALHAR-2314 Improper functioning in partitioning of sequentialFileRead property of FSRecordReaderModule.

Modified the StreamCodec to work with hashcode of filepath rather than blockId.

Conflicts:
	library/src/main/java/org/apache/apex/malhar/lib/fs/FSRecordReaderModule.java
